File: October 2024/Largest_pair_sum(8 Oct).cpp
class Solution {
  public:
    int pairsum(vector<int> &arr) {
        int mx1 = 0, mx2 = 0;
        for(int i=0; i<arr.size(); i++) {
            if(arr[i] > mx1) {
                mx2 = mx1;
                mx1 = arr[i];
            }
            else if(arr[i] > mx2) {
                mx2 = arr[i];
            }
        }
        return mx1 + mx2;
    }
};

File: October 2024/Largest Pair Sum(8 oct).java
// JAVA CODE

class Solution {
   public static int pairsum(int[] arr) {
        // code here
        int largest = Integer.MIN_VALUE, secondLargest = Integer.MIN_VALUE;
        for(int x:arr){
            if(x>largest){
                secondLargest=largest;
                largest=x;
            }
            else if(x>secondLargest)secondLargest=x;
        }
        return largest+secondLargest;
    }


}
